src/player.cpp

```diff
@@ -4121,24 +4121,21 @@ int player::overmap_sight_range(int light_level) const
     if( sight <= SEEX * 4) {
         return (sight / (SEEX / 2) );
     }
-    if ((has_amount("binoculars", 1) || has_amount("rifle_scope", 1) ||
-        has_amount("survivor_scope", 1) || -1 != weapon.has_gunmod("rifle_scope") ) &&
+    if ((has_item_with_flag("ZOOM")) &&
         !has_trait("EAGLEEYED"))  {
          if (has_trait("BIRD_EYE")) {
              return 25;
```

Comment:
What is the difference between these results here and results at line 4134? It seems that boolean check ```has_trait("EAGLEEYED")``` is now pointless and causes unnecessary duplication of code (you changed the logic of conditional statements: previously, it was ```(has_amount("binoculars", 1)``` here vs. ```!(has_amount("binoculars", 1)``` there.